eva-accession-release-automation/publish_release_to_ftp/publish_release_files_to_ftp.py

```diff
@@ -168,44 +169,49 @@ def hardlink_to_previous_release_assembly_files_in_ftp(current_release_assembly_
 
 
-def publish_assembly_release_files_to_ftp(current_release_assembly_info, release_properties):
+def publish_assembly_release_files_to_ftp(current_release_assembly_info, release_properties,
+                                          species_current_release_folder_name):
     assembly_accession = current_release_assembly_info["assembly_accession"]
-    public_release_assembly_folder = \
-        get_folder_path_for_assembly(release_properties.public_ftp_current_release_folder, assembly_accession)
+    public_release_assembly_folder = get_folder_path_for_assembly(release_properties.public_ftp_current_release_folder,
+                                                                  assembly_accession)
+    public_release_species_assembly_folder = \
+        get_folder_path_for_species_assembly(release_properties.public_ftp_current_release_folder,
+                                             species_current_release_folder_name, assembly_accession)
     # If a species was processed during this release, copy current release data to FTP
     if current_release_assembly_info["should_be_released"] and \
             current_release_assembly_info["num_rs_to_release"] > 0:
         copy_current_assembly_data_to_ftp(current_release_assembly_info, release_properties,
-                                          public_release_assembly_folder)
-    else:
-        # Since the assembly data is unchanged from the last release, hard-link instead of symlink to older release data
-        # so that deleting data in older releases does not impact the newer releases
-        # (hard-linking preserves the underlying data for a link until all links to that data are deleted)
-        hardlink_to_previous_release_assembly_files_in_ftp(current_release_assembly_info, release_properties)
+                                          public_release_assembly_folder, public_release_species_assembly_folder)
+    # Below code to hardlink to previous release commented since we will be releasing everything with new regime
+    # else:
+    #     # Since the assembly data is unchanged from the last release, hard-link instead of symlink to older release data
+    #     # so that deleting data in older releases does not impact the newer releases
+    #     # (hard-linking preserves the underlying data for a link until all links to that data are deleted)
+    #     hardlink_to_previous_release_assembly_files_in_ftp(current_release_assembly_info, release_properties)
 
     # Symlink to release README_general_info file - See layout in the link below:
     # https://docs.google.com/presentation/d/1cishRa6P6beIBTP8l1SgJfz71vQcCm5XLmSA8Hmf8rw/edit#slide=id.g63fd5cd489_0_0
-    run_command_with_output("Symlinking to release level {0} and {1} files for assembly {1}"
-                            .format(readme_general_info_file, readme_known_issues_file, assembly_accession),
+    recreate_public_release_assembly_folder(assembly_accession, public_release_assembly_folder)
```

Comment:
Not sure why this is needed, doesn't this remove the entire assembly-level folder, including the data we just copied?

Reply:
Actually as a general comment, I feel like this code would be cleaner if we used `public_release_assembly_folder` as sparingly as possible - i.e. only in the top-level method to create it if needed, and otherwise just used assembly_species_folder (`by_assembly/GCA_000005005.5/zea_mays`) and species_assembly_folder (`by_species/zea_mays/GCA_000005005.5`).  This way I'd be more confident that we're not accidentally messing up the assembly-level directory if multiple taxonomies are sharing the assembly.

Reply:
You are correct, with the changes logic, this should be created only for the first time for any assembly.
Updated it